File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
        int freq[256]={0};
        int freq1[256]={0};
        for(int i=0;i<s.size();i++){
            char s1=s[i];
            char t1=t[i];
            if(freq[s1]!=freq1[t1]){
                return false;
            }
            freq[s1]=i+1;
            freq1[t1]=i+1;
        }
        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
import java.util.*;
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int[] result =new int[256];
        int[] result1 =new int[256];
        
        for(int i=0;i<s.length();i++){
            if(result[s.charAt(i)]!=result1[t.charAt(i)]){
                return false;
            }
            result[s.charAt(i)]=i+1;
            result1[t.charAt(i)]=i+1;
        }
        return true;
    }
}